app/models/custom_field.rb

```diff
@@ -187,25 +187,40 @@ def cast_value(value)
   # Returns a ORDER BY clause that can used to sort customized
   # objects by their value of the custom field.
   # Returns false, if the custom field can not be used for sorting.
-  def order_statement
+  def order_statements
```

Comment:
Certainly not your fault as that's how the custom fields work, but there's a lot of redundancy in the queries you changed below. They actually share a lot of common knowledge. It may not be practical, but what do you think about building these queries in a helper and adding the custom constraints for specific types to the otherwise identical query?